spootify/src/common/utils/spotifyHttp.js

```diff
@@ -0,0 +1,71 @@
+import { httpGet$, httpPost$ } from "./http"
+import { of } from 'rxjs';
+import { map, switchMap, share, finalize } from 'rxjs/operators';
+
+const BASE_URL = `https://api.spotify.com/v1/browse`;
+const CLIENT_ID = process.env.REACT_APP_SPOTIFY_CLIENT_ID;
+const CLIENT_SECRET = process.env.REACT_APP_SPOTIFY_CLIENT_SECRET;
+
+let tokenExpireAt = null;
+let currentToken = null;
+let authTokenSource$ = null;
+
+const getAuthToken$ = () => {
+  const body = 'grant_type=client_credentials';
+
+  const headers = {
+    'Authorization': 'Basic ' + (new Buffer.from(CLIENT_ID + ':' + CLIENT_SECRET).toString('base64')),
+    'Content-Type': 'application/x-www-form-urlencoded',
+  }
+  return httpPost$('https://accounts.spotify.com/api/token', body, headers);
+}
```

Comment:
Get the auth token based on Spotify API doc https://developer.spotify.com/documentation/web-api/tutorials/client-credentials-flow.